### DISCUSSION TITLE
Swiper breakpoints not working for custom elements

### DISCUSSION THREAD

**Discussion by jehoshua-shey** [nolimits4web/swiper#7071] [Q&A]
I'm using SwiperJS custom elements in my NextJS project. I can seem to add breakpoints correctly case all I've done never worked including what was written in the demo code for responsive breakpoints on the SwiperJS website.

I would appreciate it if someone can put me through. I just nee breakpoints for my swipers.

**Comment by JVariance** [comment]
It works with JSON.stringify().
```js
breakpoints={
    JSON.stringify({
        768: {
            slidesPerView: 6,
            grid: {
                rows: 1,
                fill: 'row'
                }
            }
    }
)}
```


You could alternatively set the breakpoints attribute on the element when it's mounted.

```javascript
swiperContainer?.setAttribute(
    'breakpoints',
    JSON.stringify({
        768: {
            slidesPerView: 6,
            grid: {
                rows: 1,
                fill: 'row'
                }
            }
        })
    );
```

**Reply by teddarcuri** [reply]
There has to be a better solution than this. Anyone had any luck? I am also configuring my breakpoints exactly as stated in the docs, but they are not working

**Comment by YousefSaeed668** [comment]
breakpoints={{
              1024: {
                slidesPerView: 6,
              },

              768: {
                slidesPerView: 3,
              },

              0: {
                slidesPerView: 2,
              },
            }}
            i write it like this and work but once i get into the website it's broken and only appear one slide and it's a very big

**Reply by devrahuljourney** [reply]
i have same issue with exact same code. did you find any solution for making responsive. Thank you
